### ISSUE TITLE
unit test test_pdf fails on i686

### ISSUE THREAD

**Issue by toeklk** [toeklk/orocos-bayesian-filtering#11]
See http://bugs.orocos.org/show_bug.cgi?id=1064 (submitted by Till Hofmann)

test_pdf fails with the following error on i686 hardware:

test 2
    Start 2: test_pdf
2: Test command: /builddir/build/BUILD/orocos-bfl-0.8.0/tests/test_pdf
2: Test timeout computed to be: 9.99988e+06
2: DiscretePdf::Samplefrom(int, void *): No such sampling method
2: Box-Muller not implemented yet!
2: Box-Muller not implemented yet!
2: Mixture::Samplefrom(T, void *): No such sampling method
2: Mixture::Samplefrom(T, void *): No such sampling method
2: Mixture::Samplefrom(T, void *): No such sampling method
2: Mixture::Samplefrom(T, void *): No such sampling method
2: .......F.
2: 
2: pdf_test.cpp:1135:Assertion
2: Test name: PdfTest::testMixture
2: equality assertion failed
2: - Expected: 0.0590505944295115
2: - Actual  : 0.0590505944295115
2: 
2: Failures !!!
2: Run: 8   Failure total: 1   Failures: 1   Errors: 0
2/8 Test #2: test_pdf .........................***Failed    0.03 sec

Looking at the output, it seems like the test compares two floating point
numbers and expects them to be exactly equal, so this could possibly be fixed
by allowing some tolerance in the test. But that's just a guess from my side.

This was found while building a Fedora package for i686. For the complete log,
see https://kojipkgs.fedoraproject.org//work/tasks/4308/10134308/build.log

Thank you for your help!

**Comment by toeklk** [comment]
@morxa : Can you still reproduce this one using master?

**Comment by morxa** [comment]
Not exactly, but I'm having a different build error for the same reason. I'm working on a fix.
